File: fp.go
package main

import "fmt"

func increment(n int) int {
    return n + 1
}

func myObjective(arr *[5]int, f func(int)int) {
    for i := 0; i < len(arr); i++ {
        arr[i] = f(arr[i])
    }
}

func main() {
    arr := [5]int {0, 1, 2, 3, 4}
    myObjective(&arr, increment)
    
    for i := 0; i < len(arr); i++ {
        fmt.Print(arr[i], " ")
    }
}

File: fp.rs
fn increment(n: i32) -> i32 {
	n + 1
}

fn my_objective(arr: &mut [i32], f: fn(i32) -> i32) {
	for elem in arr.iter_mut() {
		*elem = f(*elem);
	}
}

fn main() {
	let mut arr: [i32; 5] = [0, 1, 2, 3, 4];
	my_objective(&mut arr, increment);

	for elem in arr.iter() {
		print!("{} ", elem);
	}
}
